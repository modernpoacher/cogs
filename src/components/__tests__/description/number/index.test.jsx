import React, { Component as mockComponent } from 'react'
import renderer from 'react-test-renderer'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import classnames from 'classnames'

import Super from 'shinkansen-cogs/components/title'
import Description from 'shinkansen-cogs/components/title/number'

Enzyme.configure({ adapter: new Adapter() })

jest.mock('classnames', () => jest.fn(() => 'MOCK CLASSNAME'))

jest.mock('shinkansen-cogs/components/title', () => ({
  __esModule: true,
  default: class MockDescription extends mockComponent {
    getClassName () { }

    renderTextContent () {}

    render () {
      const { description } = this.props

      if (description) {
        return (
          <span className={this.getClassName()}>
            {this.renderTextContent()}
          </span>
        )
      }

      return null
    }
  }
}))

describe('shinkansen-cogs/components/title/number', () => {
  describe('<Description />', () => {
    describe('With required props', () => {
      const component = (
        <Description id='MOCK ID' />
      )

      it('renders', () => {
        return expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })

      describe('`getClassName`', () => {
        it('is defined', () => {
          return expect(Description.prototype.getClassName)
            .toBeDefined()
        })
      })
    })

    describe('With additional props', () => {
      it('renders', () => {
        const component = (
          <Description
            id='MOCK ID'
            description='MOCK DESCRIPTION'
          />
        )

        return expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })
    })

    describe('`getClassName()`', () => {
      let returnValue

      beforeEach(() => {
        jest.spyOn(Super.prototype, 'getClassName').mockReturnValue('MOCK GETCLASSNAME')

        const component = (
          <Description id='MOCK ID' />
        )

        const instance = (
          shallow(component)
            .instance()
        )

        returnValue = instance.getClassName()
      })

      it('invokes `classnames`', () => {
        return expect(classnames)
          .toBeCalledWith('MOCK GETCLASSNAME', 'number')
      })

      it('returns the classname', () => {
        return expect(returnValue)
          .toBe('MOCK CLASSNAME')
      })
    })
  })
})
