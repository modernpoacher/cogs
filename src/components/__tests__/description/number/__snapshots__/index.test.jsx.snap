// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shinkansen-cogs/components/title/number <Description /> With additional props renders 1`] = `
<span
  className="MOCK CLASSNAME"
/>
`;

exports[`shinkansen-cogs/components/title/number <Description /> With required props renders 1`] = `null`;
